src/Generating/FinishGen.cpp

```diff
@@ -180,6 +180,210 @@ void cFinishGenNetherClumpFoliage::TryPlaceClump(cChunkDesc & a_ChunkDesc, int a
+	{
+		a_Output.push_back(BiomeInfo());
+	}
+
+	AStringVector ClumpInfo = StringSplitAndTrim(a_RawClumpInfo, "=");
+
+	// Information about a clump is divided in 2 parts. The biomes they can be in and the blocks that can be placed.
+	if (ClumpInfo.size() != 2)
+	{
+		LOGWARNING("OverworldClumpFoliage: Data missing for \"%s\". Please divide biome and blocks with a semi colon", a_RawClumpInfo);
+		return;
+	}
+
+	AStringVector Biomes = StringSplitAndTrim(ClumpInfo[0], ";");
+	AStringVector Blocks = StringSplitAndTrim(ClumpInfo[1], ";");
+
+	for (const auto & RawBiomeInfo : Biomes)
+	{
+		AStringVector BiomeInfo = StringSplitAndTrim(RawBiomeInfo, ",");
+		AString BiomeName = BiomeInfo[0];
+		EMCSBiome Biome = StringToBiome(BiomeName);
+		if (Biome == biInvalidBiome)
+		{
+			LOGWARNING("Biome \"%s\" is invalid.", BiomeName);
+			continue;
+		}
+
+		if (BiomeInfo.size() == 2)
+		{
+			// Only the minimum amount of clumps per chunk is changed.
+			int MinNumClump = 1;
+			if (!StringToInteger(BiomeInfo[1], MinNumClump))
+			{
+				LOGWARNING("OverworldClumpFoliage: Invalid data in \"%s\". Second parameter is either not existing or a number", RawBiomeInfo);
+				continue;
+			}
+			a_Output[Biome].m_MinNumClumpsPerChunk = MinNumClump;
+
+			// In case the minimum number is higher than the current maximum value we change the max to the minimum value.
+			a_Output[Biome].m_MaxNumClumpsPerChunk = std::max(MinNumClump, a_Output[Biome].m_MaxNumClumpsPerChunk);
+		}
+		else if (BiomeInfo.size() == 3)
+		{
+			// Both the minimum and maximum amount of clumps per chunk is changed.
+			int MinNumClumps = 0, MaxNumClumps = 1;
+			if (!StringToInteger(BiomeInfo[1], MinNumClumps) || !StringToInteger(BiomeInfo[2], MaxNumClumps))
+			{
+				LOGWARNING("Invalid data in \"%s\". Second parameter is either not existing or a number", RawBiomeInfo);
+				continue;
+			}
+
+			a_Output[Biome].m_MaxNumClumpsPerChunk = MaxNumClumps + 1;
+			a_Output[Biome].m_MinNumClumpsPerChunk = MinNumClumps;
+		}
+
+		// TODO: Make the weight configurable.
+		for (const auto & BlockName : Blocks)
+		{
+			cItem Block = cItem();
+			if (!StringToItem(BlockName, Block) && IsValidBlock(Block.m_ItemType))
+			{
+				LOGWARNING("Block \"%s\" is invalid", BlockName);
+				continue;
+			}
+
+			FoliageInfo info = FoliageInfo(static_cast<BLOCKTYPE>(Block.m_ItemType), static_cast<NIBBLETYPE>(Block.m_ItemDamage), 100);
+			a_Output[Biome].m_Blocks.push_back(info);
+		}
+	}
+}
+
+
+
+
+
+std::vector<cFinishGenClumpTopBlock::BiomeInfo> cFinishGenClumpTopBlock::ParseIniFile(cIniFile & a_IniFile, AString a_ClumpPrefix)
+{
+	std::vector<cFinishGenClumpTopBlock::BiomeInfo> foliage;
+	int NumGeneratorValues = a_IniFile.GetNumValues("Generator");
+	int GeneratorKeyId = a_IniFile.FindKey("Generator");
+	for (int i = 0; i < NumGeneratorValues; i++)
+	{
+		AString ValueName = a_IniFile.GetValueName("Generator", i);
+		if (ValueName.substr(0, a_ClumpPrefix.size()) == a_ClumpPrefix)
```

Comment:
I think the dash should be checked as well, just in case there's a new parameter in the future that starts with the prefix but does something else (`OverworldClumpFlowersMaxHeight`, for example)